### PR TITLE
OCPBUGS-34580: Add TrustedBundles to OAS container

### PR DESCRIPTION
**Pull request by jparrill**
Manual cherry pick from https://github.com/openshift/hypershift/pull/4087

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+				},
+			},
+			expectProjectedVolumeMounted: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+			tc.auditConfig.Data = map[string]string{"policy.yaml": "test-data"}
+			err := ReconcileDeployment(oapiDeployment, nil, ownerRef, testOapiCM, tc.auditConfig, nil, tc.deploymentConfig, imageName, "socks5ProxyImage", config.DefaultEtcdURL, util.AvailabilityProberImageName, false, hyperv1.AgentPlatform, tc.additionalTrustBundle, tc.clusterConf)
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go:107]: this is also sharing  oapiDeployment across test cases.
Can we please follow up to clean it up on main and understand why the other unit was failing?

**Comment by jparrill** [comment]
Follow up PR with the fixes https://github.com/openshift/hypershift/pull/4214

**Pull request merged**